testing/endtoend/evaluators/operations.go

```diff
@@ -455,3 +530,131 @@ func validatorsVoteWithTheMajority(ec *e2etypes.EvaluationContext, conns ...*grp
+	if err != nil {
+		return err
+	}
+	changes := make([]*v2.SignedBLSToExecutionChange, 0)
+	// Only send half the number of changes each time, to allow us to test
+	// at the fork boundary.
+	wantedChanges := numOfExits / 2
+	for _, idx := range exitedIndices {
+		// Exit sending more change messages.
+		if len(changes) >= wantedChanges {
+			break
+		}
+		val, err := st.ValidatorAtIndex(idx)
+		if err != nil {
+			return err
+		}
+		if val.WithdrawalCredentials[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte {
+			continue
+		}
+		if !bytes.Equal(val.PublicKey, privKeys[idx].PublicKey().Marshal()) {
+			return errors.New("pubkey is not equal")
+		}
+		message := &v2.BLSToExecutionChange{
+			ValidatorIndex:     idx,
+			FromBlsPubkey:      privKeys[idx].PublicKey().Marshal(),
+			ToExecutionAddress: bytesutil.ToBytes(uint64(idx), 20),
+		}
+		domain, err := signing.ComputeDomain(params.BeaconConfig().DomainBLSToExecutionChange, params.BeaconConfig().GenesisForkVersion, st.GenesisValidatorsRoot())
+		if err != nil {
+			return err
+		}
+		sigRoot, err := signing.ComputeSigningRoot(message, domain)
+		if err != nil {
+			return err
+		}
+		signature := privKeys[idx].Sign(sigRoot[:]).Marshal()
+		change := &v2.SignedBLSToExecutionChange{
+			Message:   message,
+			Signature: signature,
+		}
+		fmt.Printf("sending change for %d \n", idx)
+		changes = append(changes, change)
+	}
+	fmt.Printf("sending %d changes", len(changes))
+	_, err = beaconAPIClient.SubmitSignedBLSToExecutionChanges(ctx, &v2.SubmitBLSToExecutionChangesRequest{Changes: changes})
+
+	return err
+}
+
+func validatorsAreWithdrawn(ec *e2etypes.EvaluationContext, conns ...*grpc.ClientConn) error {
+	conn := conns[0]
+	beaconClient := ethpb.NewBeaconChainClient(conn)
+	debugClient := ethpb.NewDebugClient(conn)
+
+	ctx := context.Background()
+	chainHead, err := beaconClient.GetChainHead(ctx, &emptypb.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "could not get chain head")
+	}
+	stObj, err := debugClient.GetBeaconState(ctx, &ethpb.BeaconStateRequest{QueryFilter: &ethpb.BeaconStateRequest_Slot{Slot: chainHead.HeadSlot}})
+	if err != nil {
+		return errors.Wrap(err, "could not get state object")
+	}
+	versionedMarshaler, err := detect.FromState(stObj.Encoded)
+	if err != nil {
+		return errors.Wrap(err, "could not get state marshaler")
+	}
+	st, err := versionedMarshaler.UnmarshalBeaconState(stObj.Encoded)
+	if err != nil {
+		return errors.Wrap(err, "could not get state")
+	}
+
+	for key := range ec.ExitedVals {
+		valIdx, ok := st.ValidatorIndexByPubkey(key)
+		if !ok {
+			return errors.Errorf("pubkey %#x does not exist in our state", key)
+		}
+		val, err := st.ValidatorAtIndex(valIdx)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("exit epoch %d , withdrawable epoch %d \n ", val.ExitEpoch, val.WithdrawableEpoch)
+		bal, err := st.BalanceAtIndex(valIdx)
```

Comment:
balance is reduced but it would be nice to check against the correct setting in `withdrawal_credentials` or even see if the change is reflected in a block under the address it was set to.

Reply:
This can only be checked against execution, which is why I haven't included it here.